File: TinDog/src/main/java/com/polytech/tindog/Dog/DogService.java
package com.polytech.tindog.Dog;

import com.polytech.tindog.Owner.Owner;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.core.io.ByteArrayResource;
import org.springframework.core.io.Resource;
import org.springframework.stereotype.Service;
import org.springframework.web.multipart.MultipartFile;

import java.util.Date;
import java.util.UUID;

@Service
public class DogService {
    @Autowired
    private DogRepository dogRepository;

    public boolean dogExists(UUID id){
        if(dogRepository.findById(id).isPresent())
            return true;
        return false;
    }

    public Dog createDog(String dogname, Date birthday,String bio, String gender, String ownerId, MultipartFile multipartImage) throws Exception {
        Dog dog = new Dog(dogname,birthday,bio,gender,ownerId);
        dog.setPicture(multipartImage.getBytes());
        return dogRepository.save(dog);
    }

    public Resource getImage(UUID id) throws Exception {
        if(!dogExists(id))
            throw new Exception(("A dog with this id doesn't exist."));
        Dog dog = dogRepository.findById(id).get();
        return new ByteArrayResource(dog.getPicture());
    }

    public Dog[] findDogsByOwnerId(String ownerId){
        return dogRepository.findByOwnerId(ownerId).get();
    }

    public Dog getDogById(UUID id) throws Exception {
        if(!dogExists(id))
            throw new Exception(("A dog with this id doesn't exist."));
        return dogRepository.findById(id).get();
    }
}

File: TinDogUI/src/app/dog.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Dog } from './dog';
import { Owner } from './owner';

@Injectable({
  providedIn: 'root'
})
export class DogService {

  CreateDogProfile(dogname: string, birthday:Date,bio:string, gender:string, ownerId:string ,file: File): Observable<Dog>
  {
    const formData = new FormData();
    formData.append("dogname",dogname);
    formData.append("birthday", birthday.toLocaleString());
    formData.append("bio",bio);
    formData.append("gender",gender);
    formData.append("ownerId",ownerId);
    formData.append("multipartImage", file);
    return this.httpClient.post<any>("http://localhost:8080/create-dog",formData);
  }

  constructor(private httpClient: HttpClient) { }
}
